### PR TITLE
cleanup of java_types.h

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

**Review by NlightNFotis — APPROVED** [review]
Looks good to me, but I would like to know why the constructor signatures were changed from `typet` to `symbol_typet`.

File: src/java_bytecode/java_types.h [hunk 1]
```diff
@@ -83,10 +81,8 @@ std::vector<typet> java_generic_type_from_string(
 typet java_bytecode_promotion(const typet &);
 exprt java_bytecode_promotion(const exprt &);
 
-bool is_java_array_tag(const irep_idt& tag);
-
-bool is_valid_java_array(const struct_typet &type);
-
+bool is_java_array_tag(const irep_idt &tag);
+bool is_valid_java_array(const struct_typet &);
```
Comment (majakusber) [inline src/java_bytecode/java_types.h:85]: Just out of curiosity - is there a reason why the parameter name was removed from this (type) but not from the the line above (tag)?

**Review by majakusber — APPROVED** [review]
Looks good to me.

**Review by mgudemann — APPROVED** [review]
LGTM

File: src/java_bytecode/java_types.h (hunk 1 above)
Comment (kroening) [reply src/java_bytecode/java_types.h:85]: This one is borderline. The question is whether you can tell what the parameter is from the name of the function and the type of the parameter. The problem with irep_idt is that there are many uses of it; a struct_type is less ambiguous.

**Comment by kroening** [comment]
@NlightNFotis Chaging the parameter type from typet symbol_typet means that the compiler will check the precondition for you, at compile time.

**Pull request merged**